packages/backend-defaults/src/entrypoints/auditor/Auditor.ts

```diff
@@ -0,0 +1,339 @@
+      auditor = auditor.child(options.meta);
+    }
+    return new Auditor(
+      auditor,
+      {
+        auth: options?.auth,
+        httpAuth: options?.httpAuth,
+        plugin: options?.plugin,
+      },
+      redacter.add,
+    );
+  }
+
+  /**
+   * Creates a winston log formatter for redacting secrets.
+   */
+  static redacter(): {
+    format: Format;
+    add: (redactions: Iterable<string>) => void;
+  } {
+    return redacterFormat();
+  }
+
+  /**
+   * Creates a pretty printed winston log formatter.
+   */
+  static colorFormat(): Format {
+    return colorFormat();
+  }
+
+  private constructor(
+    winstonLogger: winston.Logger,
+    deps?: {
+      auth?: AuthService;
+      httpAuth?: HttpAuthService;
+      plugin?: PluginMetadataService;
+    },
+    addRedactions?: (redactions: Iterable<string>) => void,
+  ) {
+    this.#winstonLogger = winstonLogger;
+    this.#auth = deps?.auth;
+    this.#httpAuth = deps?.httpAuth;
+    this.#plugin = deps?.plugin;
+    this.#addRedactions = addRedactions;
+  }
+
+  private async log<TMeta extends JsonObject>(
+    options: AuditorEventOptions<TMeta>,
+  ): Promise<void> {
+    const auditEvent = await this.reshapeAuditorEvent(options);
+    this.#winstonLogger.info(...auditEvent);
+  }
+
+  async createEvent<TMeta extends JsonObject>(
+    options: Parameters<AuditorCreateEvent<TMeta>>[0],
+  ): ReturnType<AuditorCreateEvent<TMeta>> {
+    if (!options.suppressInitialEvent) {
+      await this.log({ ...options, status: 'initiated' });
+    }
+
+    return {
+      success: async params => {
+        // return undefined if both objects are empty; otherwise, merge the objects
+        const meta =
+          Object.keys(options.meta ?? {}).length === 0 &&
+          Object.keys(params?.meta ?? {}).length === 0
+            ? undefined
+            : { ...options.meta, ...params?.meta };
+
+        await this.log({
+          ...options,
+          meta,
+          status: 'succeeded',
+        });
+      },
+      fail: async params => {
+        // return undefined if both objects are empty; otherwise, merge the objects
+        const meta =
+          Object.keys(options.meta ?? {}).length === 0 &&
+          Object.keys(params.meta ?? {}).length === 0
+            ? undefined
+            : { ...options.meta, ...params.meta };
+
+        await this.log({
+          ...options,
+          ...params,
+          meta,
+          status: 'failed',
+        });
+      },
+    };
+  }
+
+  child(
+    meta: JsonObject,
+    deps?: {
+      auth?: AuthService;
+      httpAuth?: HttpAuthService;
+      plugin?: PluginMetadataService;
+    },
+  ): AuditorService {
+    return new Auditor(this.#winstonLogger.child(meta), {
+      auth: deps?.auth ?? this.#auth,
+      httpAuth: deps?.httpAuth ?? this.#httpAuth,
+      plugin: deps?.plugin ?? this.#plugin,
+    });
+  }
+
+  addRedactions(redactions: Iterable<string>) {
+    this.#addRedactions?.(redactions);
+  }
```

Comment:
Not sure we wan't to bother with redactions here tbh, but will have to get back to this after other changes to see what things look like

Reply:
iirc, it was done to safeguard from the potential for any secrets from being leaked.  We encountered a couple of cases before in other areas and decided it's better to be safe than sorry.